python/lsst/rapid/analysis/rubinTv.py

```diff
@@ -0,0 +1,323 @@
+from lsst.rapid.analysis.mountTorques import plotMountTracking
+from lsst.rapid.analysis.monitorPlotting import plotExp
+from lsst.atmospec.utils import isDispersedDataId
+
+CHANNELS = ["summit_imexam", "summit_specexam", "auxtel_mount_torques",
+            "auxtel_monitor"]
+
+PREFIXES = {chan: chan.replace('_', '-') for chan in CHANNELS}
+
+
+def _dataIdToFilename(channel, dataId):
+    filename = f"{PREFIXES[channel]}_dayObs_{dataId['dayObs']}_seqNum_{dataId['seqNum']}.png"
+    return filename
+
+
+def _waitForDataProduct(butler, dataProduct, dataId, logger, maxTime=20):
+    cadence = 0.25
+    maxLoops = int(maxTime//cadence)
+    for retry in range(maxLoops):
+        if butler.datasetExists(dataProduct, dataId):
+            return butler.get(dataProduct, dataId)
+        else:
+            sleep(cadence)
+    logger.warn(f'Waited {maxTime}s for {dataProduct} for {dataId} to no avail')
+    return None
+
+
+class Uploader():
+    def __init__(self):
+        if not HAS_GOOGLE_STORAGE:
+            from lsst.rapid.analysis.utils import GOOGLE_CLOUD_MISSING_MSG
+            raise RuntimeError(GOOGLE_CLOUD_MISSING_MSG)
+        self.client = storage.Client()
+        self.bucket = self.client.get_bucket('rubintv_data')
+        self.log = logging.getLogger("googleUploader")
+
+    def googleUpload(self, channel, sourceFilename, uploadAsFilename=None):
+        if channel not in CHANNELS:
+            self.log.warn(f"Error: {channel} not in {CHANNELS}")
+            return
+
+        if uploadAsFilename and (os.path.basename(sourceFilename) != uploadAsFilename):
+            finalName = os.path.join(os.path.dirname(sourceFilename), uploadAsFilename)
+            shutil.move(sourceFilename, finalName)
+            assert os.path.exists(finalName)
+        else:
+            finalName = sourceFilename
+
+        path = Path(finalName)
+        blob = self.bucket.blob("/".join([channel, path.name]))
+        self.log.info(f'Uploaded {sourceFilename} to {finalName}')
+        blob.upload_from_filename(finalName)
+
+
+class Watcher():
+    cadence = 1  # in seconds
+
+    def __init__(self, repoDir, dataProduct, **kwargs):
+        """Watch for data products in a repo.
+        """
+        self.repoDir = repoDir
+        self.butler = dafPersist.Butler(repoDir)
+        self.dataProduct = dataProduct
+        self.log = logging.getLogger("watcher")
+
+    def _getLatestExpId(self):
+        return sorted(self.butler.queryMetadata(self.dataProduct, 'expId'))[-1]
+
+    def _getDayObsSeqNumFromExpId(self, expId):
+        return self.butler.queryMetadata(self.dataProduct, ['dayObs', 'seqNum'], expId=expId)[0]
+
+    def _getLatestImageDataIdAndExpId(self):
+        expId = self._getLatestExpId()
+        dayObs, seqNum = self._getDayObsSeqNumFromExpId(expId)
+        dataId = {'dayObs': dayObs, 'seqNum': seqNum}
+        return dataId, expId
+
+    def run(self, callback, durationInSeconds=-1):
```

Comment:
I did have it subclassing at some point, but then changed back, and I can't quite remember why. Probably no good reason. It looks nice and clear like this, but I also see your point about removing the duplication. Let me give that a go...

Reply:
It's actually a bit messy because of different repo dir paths and logger names. How much do you care?

Reply:
I think you can do tricks like `self.log = log if log else Log.getLogger(__name__.partition(".")[2])` to avoid log name issues (it's what `IsrCalib` does).  In any case, it doesn't seem like it'll happen on this ticket, but keep it in mind if you end up needing to duplicate more things in each object.

Reply:
Will do, and thanks 🙂